yt_dlp/extractor/microsoftembed.py

```diff
@@ -63,3 +71,221 @@ def _real_extract(self, url):
             'subtitles': subtitles,
             'thumbnails': thumbnails,
         }
+
+
+class MicrosoftMediusBaseIE(InfoExtractor):
+    @staticmethod
+    def _sub_to_dict(subtitle_list):
+        subtitles = {}
+        for sub in subtitle_list:
+            subtitles.setdefault(sub.pop('tag', None) or 'unknown', []).append(sub)
+        return subtitles
+
+    def _extract_ism(self, ism_url, video_id):
+        formats = self._extract_ism_formats(ism_url, video_id)
+        for fmt in formats:
+            if fmt.get('language') == 'eng' or 'English' in fmt.get('format_id', ''):
+                fmt['language_preference'] = -1
+            else:
+                fmt['language_preference'] = -10
+        return formats
+
+
+class MicrosoftMediusIE(MicrosoftMediusBaseIE):
+    _VALID_URL = r'https?://medius\.microsoft\.com/[^?#]+/(?P<id>[0-9a-f\-]+)'
+
+    _TESTS = [{
+        'url': 'https://medius.microsoft.com/Embed/video-nc/9640d86c-f513-4889-959e-5dace86e7d2b',
+        'info_dict': {
+            'id': '9640d86c-f513-4889-959e-5dace86e7d2b',
+            'ext': 'ismv',
+            'title': 'Rapidly code, test and ship from secure cloud developer environments',
+            'description': 'md5:33c8e4facadc438613476eea24165f71',
+            'thumbnail': r're:https://mediusimg\.event\.microsoft\.com/video-\d+/thumbnail\.jpg.*',
+            'subtitles': 'count:30',
+        },
+        'params': {'listsubtitles': True},
+    }, {
+        'url': 'https://medius.microsoft.com/Embed/video-nc/81215af5-c813-4dcd-aede-94f4e1a7daa3',
+        'info_dict': {
+            'id': '81215af5-c813-4dcd-aede-94f4e1a7daa3',
+            'ext': 'ismv',
+            'title': 'Microsoft Build opening',
+            'description': 'md5:43455096141077a1f23144cab8cec1cb',
+            'thumbnail': r're:https://mediusimg\.event\.microsoft\.com/video-\d+/thumbnail\.jpg.*',
+            'subtitles': 'count:31',
+        },
+        'params': {'listsubtitles': True},
+    }]
+
+    def _extract_subtitle(self, webpage, video_id):
+        captions = traverse_obj(
+            self._search_json(r'const\s+captionsConfiguration\s*=\s*', webpage, 'captions', video_id, default=False),
+            ('languageList', ..., {
+                'url': ('src', {url_or_none}),
+                'tag': ('srclang', {str}),
+                'name': ('kind', {str}),
+            }))
+
+        captions = captions or traverse_obj(
```

Comment:
Does it make sense to include both, or can this lead to duplicates?
```suggestion
            })) + traverse_obj(
```

Reply:
It should be only one match exists and the existing match has all subtitles. Assuming there is a case where both matches exist, I expect there is more likely duplicate than not, since each match should have all subtitles.

Reply:
Okay, then maybe we can still combine this in to a single line with an or
```suggestion
            })) or traverse_obj(
```